internal/sorter/sorter.go

```diff
@@ -116,6 +116,17 @@ func longestRouteByHeaderAndQueryParamConditions(lhs, rhs *dag.Route) bool {
 		return len(lhs.HeaderMatchConditions) > len(rhs.HeaderMatchConditions)
 	}
 
+	// One route has a longer QueryParamMatchConditions slice.
```

Comment:
moving this up to accommodate the GW API spec also changes sort order a little since length of both header and query param matches is checked before comparing item by item

Reply:
should be fine, we only support query param matches via Gateway API anyway.